[Bug]: TS Errors When Running With `skipLibCheck: false`

### System Info


  System:
    OS: macOS 12.7.6
    CPU: (10) arm64 Apple M1 Pro
    Memory: 2.82 GB / 32.00 GB
    Shell: 5.8.1 - /bin/zsh
  Binaries:
    Node: 20.18.1 - /usr/local/bin/node
    Yarn: 3.6.4 - /opt/homebrew/bin/yarn
    npm: 10.2.5 - /opt/homebrew/bin/npm
  Browsers:
    Brave Browser: 131.1.73.101
    Safari: 15.6.1


### Details

We're building our project which has `skipLibCheck:false` in one of our build steps.

In that process, build fails with the following TS errors:

```
webiny error: ../../node_modules/@rspack/binding/binding.d.ts(172,7): error TS2380: The return type of a 'get' accessor must be assignable to its 'set' accessor type
  Type 'undefined' is not assignable to type 'string'.
../../node_modules/@rspack/core/dist/config/index.d.ts(5,1): error TS1383: Only named exports may use 'export type'.
../../node_modules/@rspack/core/dist/exports.d.ts(36,1): error TS1383: Only named exports may use 'export type'.
```

Dug into the code a bit, and for example, for the first TS2380 error, I can see it:
![Image](https://github.com/user-attachments/assets/c40815b8-ddb7-4553-a62c-d5ee1398a881)



### Reproduce link

_No response_

### Reproduce Steps

In our repo, we've run our build process with the `skipLibCheck: false` in our TS config. With this setting, we can see the above errors.

--- a/crates/node_binding/binding.d.ts
+++ b/crates/node_binding/binding.d.ts
@@ -222,7 +222,7 @@ export declare class JsModule {
   get nameForCondition(): string | undefined
   get request(): string | undefined
   get userRequest(): string | undefined
-  set userRequest(val: string)
+  set userRequest(val: string | undefined)
   get rawRequest(): string | undefined
   get factoryMeta(): JsFactoryMeta | undefined
   get type(): string

--- a/crates/rspack_binding_values/src/module.rs
+++ b/crates/rspack_binding_values/src/module.rs
@@ -135,13 +135,16 @@ impl JsModule {
   }
 
   #[napi(setter)]
-  pub fn set_user_request(&mut self, val: String) -> napi::Result<()> {
-    let module: &mut dyn Module = self.as_mut()?;
-
-    if let Ok(normal_module) = module.try_as_normal_module_mut() {
-      *normal_module.user_request_mut() = val;
-    }
-    Ok(())
+  pub fn set_user_request(&mut self, val: Either<String, ()>) -> napi::Result<()> {
+    Ok(match val {
+      Either::A(val) => {
+        let module: &mut dyn Module = self.as_mut()?;
+        if let Ok(normal_module) = module.try_as_normal_module_mut() {
+          *normal_module.user_request_mut() = val;
+        }
+      }
+      Either::B(_) => {}
+    })
   }
 
   #[napi(getter)]

--- a/crates/rspack_binding_values/src/module.rs
+++ b/crates/rspack_binding_values/src/module.rs
@@ -136,7 +136,7 @@ impl JsModule {
 
   #[napi(setter)]
   pub fn set_user_request(&mut self, val: Either<String, ()>) -> napi::Result<()> {
-    Ok(match val {
+    match val {
       Either::A(val) => {
         let module: &mut dyn Module = self.as_mut()?;
         if let Ok(normal_module) = module.try_as_normal_module_mut() {
@@ -144,7 +144,8 @@ impl JsModule {
         }
       }
       Either::B(_) => {}
-    })
+    }
+    Ok(())
   }
 
   #[napi(getter)]

--- a/packages/rspack/src/Module.ts
+++ b/packages/rspack/src/Module.ts
@@ -271,8 +271,10 @@ export class Module {
 				get(): string | undefined {
 					return module.userRequest;
 				},
-				set(val: string) {
-					module.userRequest = val;
+				set(val: string | undefined) {
+					if (val) {
+						module.userRequest = val;
+					}
 				}
 			},
 			rawRequest: {

